### ISSUE TITLE
[depth_from_video_in_the_wild] reduce_mean on masked errors

### ISSUE THREAD

**Issue by adizhol** [google-research/google-research#386]
Hi @gariel-google ,

During calculation of consistency loss, you mask the rgb and depth errors with binary mask, and the average the results:
```  
depth_error = tf.reduce_mean(
      tf.abs(frame2depth_resampled - frame1transformed_depth.depth) *
      frame1_closer_to_camera)

  rgb_error = (
      tf.abs(frame2rgb_resampled - frame1rgb) * tf.expand_dims(
          frame1_closer_to_camera, -1))
  rgb_error = tf.reduce_mean(rgb_error)
```
But  averaging on all the error "pixels" reduces the loss. Wouldn't averaging only the valid (masked) error values be more correct?

Thank you,
Adi

**Comment by bolianchen** [comment]
Hi @adizhol 

We acknowledged what you mentioned and add --use_weighted_l1 option to only average over effective pixels. You may be interested to check the repo: 
https://github.com/bolianchen/pytorch_depth_from_videos_in_the_wild

        self.parser.add_argument('--use_weighted_l1',
                                 action = 'store_true',
                                 help='use weighted_l1 error which would not'
                                      ' include the masked out regions into '
                                      'the denominator')
Best,
Bolian
